Ex4.java

```diff
@@ -0,0 +1,47 @@
+public class Ex4 {
+
+    /**
+     * The fuction loop over the mapTiles array and count the number of islands
+     * @param numOfRow
+     * @param numOfCol
+     * @param mapTiles - 2-dimensional array of booleans where false means water and true means land
+     * @return
+     */
+    public static int countIslands(int numOfRow, int numOfCol, boolean[][] mapTiles) {
+        int numOfIslands = 0;
+        boolean[][] visit = new boolean[numOfRow][numOfCol];
+        for (int i = 0; i < numOfRow; i++) {
+            for (int j = 0; j < numOfCol; j++) {
+                if (!visit[i][j] && mapTiles[i][j]) {
+                    numOfIslands++;
+                    // find all the tiles that belong to this island
+                    findAllTilesOfIsland(mapTiles, visit, numOfRow, numOfCol, i, j);
+                }
+            }
+        }
+        return numOfIslands;
+    }
+
+    /**
+     * The function check if the arguments legal and check

```

Comment:
Typically you do not need to write that method checks arguments. Just explain its main functionality, i.e. finding all tiles that belong to the island containing tile `(rowIndex, colIndex)` and marking them as visited.